### DISCUSSION TITLE
Dependecy Injection IHttpContextAccessor

### DISCUSSION THREAD

**Discussion by MarioEME** [danielgerlag/workflow-core#1027] [Q&A] [answered]
Hello,

I'm using workflow-core in an asp.net core (mvc) project.
In one of my steps I need to get the user that was logged in when the workflow was called, so I'm injecting IHttpContextAccessor in the step to get the user in the httpContextAccessor.HttpContext  but this property is always null.
Any idea why?

Heres a sample code:

 public class FindCreator : StepBodyAsync
 {
        public string OutCreator { get; set; }
        public Request InRequest { get; set; }

        private IRolesService _roleService;
        private IHttpContextAccessor _httpContextAccessor, 

        public FindCreator(IHttpContextAccessor httpContextAccessor, IRolesService roleService)
        {
            _roleService = roleService;
            _httpContextAccessor = httpContextAccessor;
        }
}

Thanks

**Comment by Alundra828** [comment]
I'm also running into this problem. Could it be that the workflows operate on a thread that can't access the httpcontext?

@MarioEME , have you found a solution in the months since you posted?

**Comment by danielgerlag** [comment, marked as answer]
The workflow does not  execute within the scope of the original call... it could be months later or even on a different machine.... you will need to capture the info you need before you create the workflow and attached it to the workflow data.
